src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolTargetingNonSupportedTFM.cs

```diff
@@ -0,0 +1,76 @@
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.Build.Tasks;
+using System;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolTargetingNonSupportedTFM : SdkTest
+    {
+
+        public GivenThatWeWantToPackAToolTargetingNonSupportedTFM(ITestOutputHelper log) : base(log)
+        {
+
+        }
+
+        [Theory]
+        // lower than netcoreapp2.0
+        [InlineData("TargetFramework", "netcoreapp2.0", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        [InlineData("TargetFramework", "netcoreapp1.1", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        [InlineData("TargetFrameworks", "netcoreapp2.0;netcoreapp2.1", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        // non netcoreapp
+        [InlineData("TargetFramework", "netstandard2.0", "PackAsToolOnlySupportNetcoreapp")]
+        public void It_should_fail_with_error_message(string targetFrameworkProperty, string targetFramework, string error)
+        {
+            TestAsset helloWorldAsset = _testAssetsManager
+                                        .CopyTestAsset("PortableTool", "PackNonSupportedTFM")
+                                        .WithSource()
+                                        .WithProjectChanges(project =>
+                                        {
+                                            XNamespace ns = project.Root.Name.Namespace;
+                                            XElement propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+
+                                            propertyGroup.Element(ns + "TargetFramework").Remove();
+                                            propertyGroup.Add(new XElement(ns + targetFrameworkProperty, targetFramework));
+
+                                        })
+                                        .Restore(Log);
+            var packCommand = new PackCommand(Log, helloWorldAsset.TestRoot);
+
+            CommandResult result = packCommand.Execute();
+            result.ExitCode.Should().NotBe(0);
+
+            // walk around attribute requires static
+            string message = null;
+            if (error == "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")
+            {
+                message = Strings.PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21;
+            }
+            else if (error == "PackAsToolOnlySupportNetcoreapp")
+            {
+                message = Strings.PackAsToolOnlySupportNetcoreapp;
+            }
+            else
+            {
+                throw new ArgumentException("No matching error to assert");
+            }
+            result.StdOut.Should().Contain(message);
+        }
+
+        [FullMSBuildOnlyFact]
+        public void It_should_fail_with_error_message_on_fullframework()
+        {
+            It_should_fail_with_error_message("TargetFramework", "net46", "PackAsToolOnlySupportNetcoreapp");
```

Comment:
I am blocking full framework. It should not work in the beginning

Reply:
@KathleenDollard I am blocking full framework and netstandard at the same time.

Reply:
Yes, let's notify on pack if it will fail later when we can. 

I don't think I've seen a message on netstandard.

Reply:
No, but it is pretty much impossible to create one. There is no entry point for the program at all. It will fail later down the road, but in this case it can fail more explicitly with better error message.